apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -38,8 +40,10 @@ where
         schema_id: Option<String>,
         config: &crate::configuration::QueryPlanning,
     ) -> CachingQueryPlanner<T> {
-        let cache =
-            Arc::new(DeduplicatingCache::from_configuration(&config.experimental_cache).await);
+        let cache = Arc::new(
+            DeduplicatingCache::from_configuration(&config.experimental_cache, "query planner")
```

Comment:
What's the meaning of `"query planner"` here ? Is it ok to have space inside ?

Reply:
it is used here: https://github.com/apollographql/router/pull/2176/files#diff-0da006a7a98a66bf432b9e5759268b9339ce0633b44891cd29004870742069d8R67

if we cannot connect to Redis, the error message indicates in which cache it saw the issue (because each of them could have different redis configs).
I chose to let the router start if we cannot connect to redis, but I could change it so that it does not start unless we can connect. That would imply adding connection checks to redis very early in the process, which I wanted to avoid